platform/android/os_android.cpp

```diff
@@ -246,11 +246,11 @@ bool OS_Android::is_mouse_grab_enabled() const {
 }
 
 Point2 OS_Android::get_mouse_position() const {
-	return hover_prev_pos;
+	return input->get_mouse_position();
 }
 
 int OS_Android::get_mouse_button_state() const {
-	return buttons_state;
+	return input->get_mouse_button_mask();
 }
```

Comment:
How are the mouse position and the mouse button mask updated?

Reply:
`InputDefault` tracks them as well, and in a more comprehensive way. My idea is to keep `AndroidInputHandler`s own tracking by now, and in the future, when we remove OS-specific tracking from everywhere, we bring that last amount of sanity here.